process/block/interceptedBlocks/common.go

```diff
@@ -101,11 +103,48 @@ func checkMetaShardInfo(shardInfo []data.ShardDataHandler, coordinator sharding.
 		if err != nil {
 			return err
 		}
+
+		err = checkProofs(sd)
+		if err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+func checkProofs(shardData data.ShardDataHandler) error {
+	err := checkProof(shardData.GetPreviousProof())
+	if err != nil {
+		return fmt.Errorf("%w for previous block", err)
+	}
+
+	err = checkProof(shardData.GetCurrentProof())
```

Comment:
I would suggest to remove the current proof from the sharddata
the proof is anyway checked when making sure the headers that generated the shard data are available and have proofs.